Credentials/src/CredentialsCredentialGateway.php

```diff
@@ -0,0 +1,190 @@
+                'gibbonPerson.preferredName', 
+                'gibbonYearGroup.nameShort AS yearGroup', 
+                'gibbonRollGroup.nameShort AS rollGroup', 
+                '(SELECT COUNT(gibbonPersonID) FROM credentialsCredential WHERE gibbonPersonID=gibbonPerson.gibbonPersonID) AS credentialCount'
+            ])
+            ->leftJoin('gibbonStudentEnrolment', 'gibbonPerson.gibbonPersonID=gibbonStudentEnrolment.gibbonPersonID AND gibbonStudentEnrolment.gibbonSchoolYearID = :gibbonSchoolYearID')
+            ->leftJoin('gibbonYearGroup', 'gibbonStudentEnrolment.gibbonYearGroupID=gibbonYearGroup.gibbonYearGroupID')
+            ->leftJoin('gibbonRollGroup', 'gibbonStudentEnrolment.gibbonRollGroupID=gibbonRollGroup.gibbonRollGroupID')
+            ->where("(SELECT COUNT(gibbonPersonID) FROM credentialsCredential WHERE gibbonPersonID=gibbonPerson.gibbonPersonID) > 0")
+            ->bindValue('gibbonSchoolYearID', $gibbonSchoolYearID);
+
+        if ($criteria->hasFilter('all')) {
+            $query->innerJoin('gibbonRole', 'FIND_IN_SET(gibbonRole.gibbonRoleID, gibbonPerson.gibbonRoleIDAll)')
+                  ->where("gibbonRole.category='Student'");
+        } else {
+            $query->where("gibbonStudentEnrolment.gibbonStudentEnrolmentID IS NOT NULL")
+                  ->where("gibbonPerson.status = 'Full'")
+                  ->where('(gibbonPerson.dateStart IS NULL OR gibbonPerson.dateStart <= :today)')
+                  ->where('(gibbonPerson.dateEnd IS NULL OR gibbonPerson.dateEnd >= :today)')
+                  ->bindValue('today', date('Y-m-d'));
+        }
+
+        $criteria->addFilterRules($this->getSharedUserFilterRules());
+
+        return $this->runQuery($query, $criteria);
+    }    
+
+    public function queryViewCredentialsByPerson(QueryCriteria $criteria, $gibbonPersonID) {
+        $query = $this
+                ->newQuery()
+                ->from($this->getTableName())
+                ->cols([
+                    'credentialsCredential.credentialsCredentialID',
+                    'credentialsCredential.username',
+                    'credentialsCredential.password',
+                    'credentialsCredential.notes as credentialNotes',
+                    'credentialsWebsite.title',
+                    'credentialsWebsite.logo',
+                    'credentialsWebsite.notes as websiteNotes',
+                    'credentialsWebsite.url'
+                ])
+                ->innerJoin('credentialsWebsite', 'credentialsCredential.credentialsWebsiteID=credentialsWebsite.credentialsWebsiteID')
+                ->where("credentialsWebsite.active = 'Y'")
+                ->where("gibbonPersonID=:gibbonPersonID")
+                ->orderBy(['title'])
+                ->bindValue('gibbonPersonID', $gibbonPersonID);     
+        return $this->runQuery($query, $criteria);
+    }
+    
+    public function selectCredentialsCredentialById($credentialsCredentialID) {
```

Comment:
These could be swapped for the built-in `getByID`, `insert`, `update`, `delete' methods.